tools/testsys/src/crds.rs

```diff
@@ -0,0 +1,405 @@
+use crate::error::{self, Result};
+use crate::run::{KnownTestType, TestType};
+use bottlerocket_types::agent_config::TufRepoConfig;
+use bottlerocket_variant::Variant;
+use log::{debug, warn};
+use maplit::btreemap;
+use model::test_manager::{SelectionParams, TestManager};
+use model::Crd;
+use pubsys_config::RepoConfig;
+use snafu::OptionExt;
+use std::collections::BTreeMap;
+use testsys_config::{rendered_cluster_name, GenericVariantConfig, TestsysImages};
+
+/// A type that is used for the creation of all CRDs.
+pub struct CrdInput<'a> {
+    pub client: &'a TestManager,
+    pub arch: String,
+    pub variant: Variant,
+    pub config: GenericVariantConfig,
+    pub repo_config: RepoConfig,
+    pub starting_version: Option<String>,
+    pub migrate_to_version: Option<String>,
+    /// `CrdCreator::starting_image_id` function should be used instead of using this field, so
+    /// it is not externally visible.
+    pub(crate) starting_image_id: Option<String>,
+    pub images: TestsysImages,
+}
+
+impl<'a> CrdInput<'a> {
+    /// Retrieve the TUF metadata from `Infra.toml`
+    pub fn tuf_metadata(&self) -> Option<TufRepoConfig> {
+        if let (Some(metadata_base_url), Some(targets_url)) = (
+            &self.repo_config.metadata_base_url,
+            &self.repo_config.targets_url,
+        ) {
+            debug!(
```

Comment:
Can we enable debug log level through `cargo make` currently? Aside: it might be nice to have a globally respected log level for Bottlerocket build and test processes that is respected throughout `Makefile.toml`.

Reply:
`-e RUST_LOG=debug` can be used to see those logs.